packages/rulesets/src/spectral/functions/Extensions/next-link-property-must-exist.ts

```diff
@@ -0,0 +1,27 @@
+// An x-ms-pageable extension passes this rule if the value for nextLinkName refers to a string property
+// that exists on the schema for the 200 response of the parent operation.
+
+export const nextLinkPropertyMustExist = (opt: any, _opts: any, ctx: any) => {
+  if (opt === null || typeof opt !== "object") {
+    return [];
+  }
+  if (opt["x-ms-pageable"] === undefined) {
+    return [];
+  }
+  const path = ctx.path || [];
+  const errors: any = [];
+  const nextLinkName = opt["x-ms-pageable"]?.nextLinkName || null;
+  const responseSchemaProperties = opt?.responses?.["200"]?.schema?.properties || {};
```

Comment:
need to use [getProperties](https://github.com/Azure/azure-openapi-validator/blob/9ff0d6c959d29df4fbd34e7a2163584850d3bba1/packages/rulesets/src/spectral/functions/utils.ts#L4) to get all properties , it will include the properties from the allOf models